crates/cxx-qt-gen/src/naming/type_names.rs

```diff
@@ -178,6 +178,15 @@ impl TypeNames {
     }
 
     /// For a given rust ident return the CXX name with its namespace
+    ///
+    /// Ideally we'd want this type name to always be **fully** qualified, staring with `::`.
+    /// Unfortunately, this isn't always possible, as the Qt5 meta object system doesn't register
+    /// types with the fully qualified path :(
+    /// E.g. it will recognize `QString`, but not `::QString` from QML.
```

Comment:
hmm what we could possibly do, any type that is registered in a C++ block we could do the required Qt registration in our C++ generation :thinking: 

Eg if there is a `type T = ...` then we could generate the missing type declaration with the fully qualified version. Guess the only fun would be if the type is not in fact supposed to be registered?

Reply:
Interesting idea...
At least we could do that for all QObject types that we found in `extern "C++Qt"` and `extern "RustQt"`.
Though as you said, unsure if there's ever a point where it **shouldn't** be registered :thinking: 

Maybe let's add this as a note into #619 and try this in another PR.

Reply:
Commented :-)